backend/server.js

```diff
@@ -61,17 +62,19 @@ hapi.ext({
 sbp('okTurtles.data/set', SERVER_INSTANCE, hapi)
 
 sbp('sbp/selectors/register', {
-  'backend/server/broadcastEntry': async function (entry: GIMessage) {
+  'backend/server/broadcastEntry': async function (deserializedHEAD: Object, entry: string) {
     const pubsub = sbp('okTurtles.data/get', PUBSUB_INSTANCE)
-    const pubsubMessage = createMessage(NOTIFICATION_TYPE.ENTRY, entry.serialize())
-    const subscribers = pubsub.enumerateSubscribers(entry.contractID())
-    console.debug(chalk.blue.bold(`[pubsub] Broadcasting ${entry.description()}`))
+    const pubsubMessage = createMessage(NOTIFICATION_TYPE.ENTRY, entry)
+    const subscribers = pubsub.enumerateSubscribers(deserializedHEAD.contractID)
+    console.debug(chalk.blue.bold(`[pubsub] Broadcasting ${deserializedHEAD.description()}`))
     await pubsub.broadcast(pubsubMessage, { to: subscribers })
   },
-  'backend/server/handleEntry': async function (entry: GIMessage) {
-    sbp('okTurtles.data/get', PUBSUB_INSTANCE).channels.add(entry.contractID())
-    await sbp('chelonia/db/addEntry', entry)
-    await sbp('backend/server/broadcastEntry', entry)
+  'backend/server/handleEntry': async function (deserializedHEAD: Object, entry: string) {
+    sbp('okTurtles.data/get', PUBSUB_INSTANCE).channels.add(deserializedHEAD.contractID)
+    await sbp('chelonia/private/in/enqueueHandleEvent', deserializedHEAD.contractID, entry)
+    // Persist the Chelonia state after processing a message
+    await sbp('chelonia/db/set', '_private_cheloniaState', JSON.stringify(sbp('chelonia/private/state')))
```

Comment:
As this value will get very large, we don't want to write a massive amount of data to disk each and every time we get a new message. So instead, we need to split up the state under multiple keys, one per contract.

Reply:
Done. We can't really use `head=` because it's updated at a different time and it holds different information.